tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -37,6 +38,89 @@
 #define DEFAULT_RETRY_PERIOD                                                                       \
 	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("5 min"), InvalidOid, -1))
 
+typedef struct DropChunksMeta
+{
+	Hypertable *ht;
+	Oid ht_oid;
+	FormData_ts_interval *older_than;
+	bool is_matht; /* is this a materialized hypertable associated
+					  with a continuous aggregate */
+} DropChunksMeta;
+
+static void
+validate_drop_chunks_hypertable(Cache *hcache, Oid user_htoid, Oid older_than_type,
+								Datum older_than_datum, DropChunksMeta *meta)
+{
+	FormData_ts_interval *older_than;
+	ContinuousAggHypertableStatus status;
+
+	meta->ht = NULL;
+	meta->ht_oid = user_htoid;
```

Comment:
I would move initialization of this struct outside this function. It seems that now you just need to pass in the struct and the parameters to fill in instead of just passing in the struct. Instead, the caller can do:

```
DropChunksMeta meta = {
   .ht_oid = user_htoid,
   .hcache = ts_hypertable_cache_pin(),
};
```

and avoid passing in the extra parameters.